pkg/cache/clusterqueue.go

```diff
@@ -202,18 +209,43 @@ func (c *ClusterQueue) UpdateRGByResource() {
 	}
 }
 
-// UpdateWithFlavors updates a ClusterQueue based on the passed ResourceFlavors set.
-// Exported only for testing.
-func (c *ClusterQueue) UpdateWithFlavors(flavors map[kueue.ResourceFlavorReference]*kueue.ResourceFlavor) {
+func (c *ClusterQueue) updateQueueStatus() {
 	status := active
-	if flavorNotFound := c.updateLabelKeys(flavors); flavorNotFound {
+	if c.hasMissingFlavors || c.hasMissingAdmissionChecks {
 		status = pending
 	}
-
-	if c.Status != terminating {
+	if c.Status == terminating {
+		status = terminating
+	}
+	if status != c.Status {
 		c.Status = status
+		metrics.ReportClusterQueueStatus(c.Name, c.Status)
+	}
+}
+
+func (c *ClusterQueue) inactiveReason() (string, string) {
+	switch c.Status {
+	case terminating:
+		return "Terminating", "Can't admit new workloads; clusterQueue is terminating"
+	case pending:
+		switch {
+		case c.hasMissingFlavors && c.hasMissingAdmissionChecks:
+			return "FlavorAndCheckNotFound", "Can't admit new workloads; some flavors and admission checks are not found"
```

Comment:
why not separate messages for each?

If not, it should be `Or` instead of `And`.

Reply:
Ah, sorry, I see what you are doing now.